### DISCUSSION TITLE
Change memory device name

### DISCUSSION THREAD

**Discussion by jojo2357** [nokyan/resources#205] [Ideas]
I'm not sure if this deserves an issue without a conversation first. I am looking at the memory device name (24.00 GiB DDR4 in the image below)

![image](https://github.com/nokyan/resources/assets/66704796/91af8335-7af5-4f54-8e7d-190847072d0e)
![image](https://github.com/nokyan/resources/assets/66704796/0a959fea-0e1e-4f4e-98d3-e1d1cfa9cc67)

I have no idea where these numbers come from, and why they disagree with the number ***right below it***

I did my due diligence and traced the source of that number to `DMIDECODE` from the command `dmidecode -t 17 -q`

Pertinent output of that command is below:
```
	Size: 8 GB
	Size: 16 GB
```

However, cating `/proc/meminfo` gives `MemTotal` at `24521520 k(i)B` for `23.39 GiB` of total memory, or `25.11 GB` of memory.

From `man dmidecode`:
> **BUGS**
       More often than not, information contained in the DMI tables is inaccurate, incomplete or simply wrong.

## My Suggestion
Just add the reported memory (8 + 16 = 24) and do not convert between `GiB` and `GB` so in my case it will just say 24GB of ram no matter what. That is simple, that is what I tell people I have in my device.

Or, add up swap and memory to give a total amount of memory in the machine, perhaps from `/proc/meminfo` instead of `dmidecode`?

Maybe I've missed something. Just an idea.

**Comment by nokyan** [comment]
Hi,
sorry for responding this late, I didn't have notifications turned on for discussions.

I'm using dmidecode's output for the device description string mainly because I needed something for memory's device description string and I didn't just want to repeat the exact same thing as for the usage string. dmidecode's output has the advantage of (in my experience) counting not just the memory available to the system (what /proc/meminfo says) but the actual amount of RAM installed on your mainboard. If you use an integrated GPU, a certain part of your RAM will be allocated to and solely usable by the integrated GPU and is thus not counted towards the memory available to the system.


> From `man dmidecode`:
>
>  >  **BUGS**
>  >  More often than not, information contained in the DMI tables is inaccurate, incomplete or simply wrong.

I didn't know about that and frankly I haven't had any issues with dmidecode's reliability before. I also don't recall any issues in this repo about wrong information by dmidecode that was dmidecode's fault.

> Just add the reported memory (8 + 16 = 24) and do not convert between GiB and GB so in my case it will just say 24GB of ram no matter what. That is simple, that is what I tell people I have in my device.

I understand what you mean but I'd prefer the way it currently works just because it's more predictable. When I change the data unit prefix, I expect *every* data-based value to change, not just most of them with exceptions.

> add up swap and memory to give a total amount of memory in the machine

I think this could be confusing, though how about instead showing them both seperated in the device description string? Like "32.00 GiB DDR4 · 8.00 GiB Swap". That could make it more useful I think.
